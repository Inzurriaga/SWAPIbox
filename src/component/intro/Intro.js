import React, { Component } from "react"

export default class Intro extends Component{
    constructor(){
        super();
        this.state = {
            title: "",
            bodytext: ""
        }
    }

    componentDidMount = () => {
        const randomNum = Math.ceil(Math.random() * 7)
        const url = `https://swapi.co/api/films/${randomNum}`
        fetch(url)
            .then(response => response.json())
            .then(films => this.setState({
                title: films.title,
                bodytext: films.opening_crawl
            }))
    }

    render = () => {
        const { title, bodytext } = this.state
        return(
            <div>
                <button onClick={this.props.exitIntro}>asdfas</button>
                <h1>{title}</h1>
                <p>{bodytext}</p>
            </div>
        )
    }
}