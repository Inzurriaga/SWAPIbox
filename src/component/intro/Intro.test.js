import React from "react"
import { shallow } from "enzyme"
import Intro from "./Intro"

describe("intro", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <Intro />
        )
    })
    it("should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()
    })
    it("should have a default state", () => {
        expect(wrapper.state()).toEqual({
            title: "",
            bodytext: ""
        })
    })
})