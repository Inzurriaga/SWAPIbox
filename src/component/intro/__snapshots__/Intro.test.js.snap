// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`intro should match snapshot 1`] = `
<section
  className="intro-section"
>
  <button>
    x
  </button>
  <div
    className="fade"
  />
  <div
    className="crawl-text"
  >
    <div
      className="crawl"
    >
      <h1 />
      <p />
    </div>
  </div>
</section>
`;
